src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,256 @@
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinRecord.cost_records =
+            // @ts-ignore
+            allCostList.filter(cost => cost.clin === clinRecord.clin_number &&
+              cost.task_order_number === taskOrder.task_order_number); // FIXME temp code
+          // allCostList.filter(cost => cost.clin?.value === clinRecord.sys_id);//FIXME correct code
+        })
+      })
+    })
+  }
+
+  /**
+   * Performs the 'Total Obligated' and 'Funds Spent' for all the portfolios in the list
+   *
+   * NOTE: Instead of computing here if we were to make the aggregate API call, it would take
+   * as many calls as the number of portfolios to get, for example, the 'Total Obligated' from
+   * the CLINS table. The reason for this is, the aggregate query response does not have any
+   * reference to what portfolio the aggregate belongs to and so a call for each portfolio.
+   * Computing here will eliminate all these calls.
+   */
+  @Action({rawError: true})
+  private computeAllAggregations(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.dod_component = 'ARMY' // FIXME: delete this line after API starts returning
+      let totalObligatedForPortfolio = 0;
+      let totalFundsSpentForPortfolio = 0;
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          if (clinRecord.clin_status === 'ACTIVE' ||
+            clinRecord.clin_status === 'OPTION_EXERCISED') { // TODO: double check the statuses
+            totalObligatedForPortfolio =
+              totalObligatedForPortfolio + Number(clinRecord.funds_obligated);
+          }
+          clinRecord.cost_records?.forEach(costRecord => {
+            totalFundsSpentForPortfolio = totalFundsSpentForPortfolio + Number(costRecord.value);
+          })
+        })
+      })
+      portfolio.funds_obligated = totalObligatedForPortfolio;
+      portfolio.funds_spent = totalFundsSpentForPortfolio;
+    })
+  }
+
+  @Action({rawError: true})
+  public async loadPortfolioSummaryList(loggedInUserId: string): Promise<PortfolioSummaryDTO[]> {
+    await this.ensureInitialized();
+    try {
+      const query =
+        `portfolio_managersLIKEe0c4c728875ed510ec3b777acebb356`; // TODO use loggedInUserId
+      const portfolioSummaryListRequestConfig: AxiosRequestConfig = {
+        params: {
+          // sysparm_fields: fields, // ??? does not passing this return all fields?
+          sysparm_query: query
+        },
+      };
+      const portfolioSummaryList =
+        await api.portfolioTable.getQuery(portfolioSummaryListRequestConfig);
+      if (portfolioSummaryList && portfolioSummaryList.length > 0) {
+        // call out to set the cloud service provider
+        await this.setCspDisplay(portfolioSummaryList);
+        // todo: call outs to other functions to set data from other tables
```

Comment:
is this still a `todo` ? I think all of the calls are being made, right?

Reply:
Not a todo any more. Merged that with the other comment on line 235.